web-ui/src/main/resources/catalog/components/common/map/mapService.js

```diff
@@ -1432,6 +1442,34 @@
             return defer.promise;
           },
 
+          updateWmsLayerForProj: function(map, layer, proj) {
+            var $this = this;
+
+            var layerUrl = layer.get('url');
+            var layerName = layer.get('name');
+            if (!layerUrl || !layerName) {
+              return;
+            }
+
+            // Remove the layer from the map
+            map.removeLayer(layer);
+
+            $this.checkProj4Def(proj.getCode())
+              .then(function (isCustomDef) {
+                gnOwsCapabilities.getWMSCapabilities(layerUrl)
+                  .then(function(capObj) {
+                    capObj.layers.forEach(function (lyr) {
+                      if (lyr.Name !== layerName) {
+                        return;
+                      }
+                      
+                      var style = layer.getSource().getParams().STYLES;
+                      $this.addWmsToMapFromCap(map, lyr, style);
```

Comment:
According to the OpenLayers examples and docs, it should be enough to call `layer.getSource().updateParams()` to redraw the WMS using the new projection. Unfortunately, that didn't seem to do anything, which is why we're removing and re-adding the layer here.